client/asset/interface.go

```diff
@@ -148,14 +155,14 @@ type Wallet interface {
 	//
 	// NOTE: This could potentially be a long and expensive operation if performed
 	// long after the swap is broadcast; might be better executed from a goroutine.
-	FindRedemption(ctx context.Context, coinID dex.Bytes) (redemptionCoin, secret dex.Bytes, err error)
+	FindRedemption(ctx context.Context, coinID, contract dex.Bytes) (redemptionCoin, secret dex.Bytes, err error)
 	// Refund refunds a contract. This can only be used after the time lock has
 	// expired AND if the contract has not been redeemed/refunded.
 	// NOTE: The contract cannot be retrieved from the unspent coin info as the
 	// wallet does not store it, even though it was known when the init transaction
 	// was created. The client should store this information for persistence across
 	// sessions.
-	Refund(coinID, contract dex.Bytes) (dex.Bytes, error)
+	Refund(coinID, contract dex.Bytes, startTime time.Time) (dex.Bytes, error)
```

Comment:
Would you mind just reviewing the comments for the modified `Wallet` interface functions and add a little bit about the time and contract args please, especially where it's not obvious why they are required? e.g. it's not clear why `Refund` needs the time, but `FindRedemption` does not, or if a `Wallet` implementation may or may not require the values depending on if they are spv or full-node backed.

Because an `interface` defines an API but also the _intent_ of a concrete implementation, I feel it would be helpful to describe the args in terms like *may*, *should*, and *must* so that the person implementing understands when and if an arg is not relevant or how it should be considered.